presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -270,7 +270,7 @@ private void rehash()
 
         long newCapacityLong = hashCapacity * 2L;
         if (newCapacityLong > Integer.MAX_VALUE) {
-            throw new PrestoException(GENERIC_INSUFFICIENT_RESOURCES, "Size of hash table cannot exceed 1 billion entries");
```

Comment:
The old comment is correct. 1B is the largest number of entries we support. Anything above that fails.

I understand that the original message is tricky. But it's also a closer bound.

If you have some nice ideas, it will be great.